draft-west-cookie-incrementalism.md

```diff
@@ -224,6 +233,48 @@ attribute (Section 4.1.2.5 of {{RFC6265bis}}) by altering the storage model defi
 This is conceptually similar to the requirements put into place for the `__Secure-` prefix (Section
 4.1.3.1 of {{RFC6265bis}}).
 
+## Schemeful Same-Site {#schemeful-samesite}
+
+By using the scheme, as well as the registrable domain, schemeful same-site can
+help to protect https origins against a network attacker that is impersonating
+an http origin with the same registrable domain, thereby further increasing
+SameSite's CSRF protections. To do so we need to modify a number of things:
+
+First change the definition of "site for cookies" from a registrable domain to
+an origin. In the places where a we return an empty string for a non-existent
+"site for cookies" we should instead return an origin set to a freshly
+generated globally unique identifier.
+
+Then replace the same-site calculation algorithm with the following:
+
+~~~
+Two origins, A and B, are considered same-site if the following algorithm returns true:
+1.  If A and B are both scheme/host/port triples then
+
+    1.  If A's scheme does not equal B's scheme, return false.
+
+    2.  Let hostA be A's host, and hostB be B's host.
+
+        1.  If hostA equals hostB and hostA's registrable domain is null, return true.
+
+        2.  If hostA's registrable domain equals hostB's registrable domain and is non-null, return true.
+
+2.  If A and B are both the same globally unique identifier, return true.
```

Comment:
As above, I'd like to shift towards "opaque origin".

Reply:
See my above comment.